docs/usage/deriving-data-selectors.md

```diff
@@ -0,0 +1,624 @@
+
+// Function declaration, mapping over an array to derive values
+function selectItemIds(state) {
+  return state.items.map(item => item.id)
+}
+
+// Function declaration, encapsulating a deep lookup
+function selectSomeSpecificField(state) {
+  return state.some.deeply.nested.field
+}
+
+// Arrow function, deriving values from an array
+const selectItemsWhoseNamesStartWith = (items, namePrefix) =>
+  items.filter(item => item.name.startsWith(namePrefix))
+```
+
+A selector function can have any name you want. However, [**we recommend prefixing selector function names with the word `select` combiend with a description of the value being selected**](../style-guide/style-guide.md#name-selector-functions-as-selectthing). Typical examples of this would look like **`selectTodoById`**, **`selectFilteredTodos`**, and **`selectVisibleTodos**`.
+
+If you've used [the `useSelector` hook from React-Redux](../tutorials/fundamentals/part-5-ui-and-react.md), you're probably already familiar with the basic idea of a selector function - the functions that we pass to `useSelector` must be selectors:
+
+```js
+function TodoList() {
+  // highlight-start
+  // This anonymous arrow function is a selector!
+  const todos = useSelector(state => state.todos)
+  // highlight-end
+}
+```
+
+Selector functions are typically defined in two different parts of a Redux application:
+
+- In slice files, alongside the reducer logic
+- In component files, either outside the component, or inline in `useSelector` calls
+
+A selector function can be used anywhere you have access to the entire Redux root state value. This includes the `useSelector` hook, the `mapState` function for `connect`, middleware, thunks, and sagas.
+
+It's not typically possible to use selectors inside of reducers, because a slice reducer only has access to its own slice of the Redux state, and most selectors expect to be given the _entire_ Redux root state as an argument.
+
+### Encapsulating State Shape with Selectors
+
+The first reason to use selector functions is for encapsulation and reusability when dealing with your Redux state shape.
+
+Let's say that one of your `useSelector` hooks makes a very specific lookup into part of your Redux state:
+
+```js
+const data = useSelector(state => state.some.deeply.nested.field)
+```
+
+That's a legal statement. But, imagine that you've got several components that need to access that field. What happens if you need to make a change to where that piece of state lives? You would now have to go change _every_ `useSelector` hook that references that value. So, in the same way that [we recommend using action creators to encapsulate details of creating actions](../style-guide/style-guide.md#use-action-creators), we recommend defining reusable selectors to encapsulate the knowledge of where a given piece of state lives. Then, you can use a given selector function many times in the codebase, anywhere that your app needs to retrieve that particular data.
```

Comment:
```suggestion
That's a valid selector. But, imagine that you've got several components that need to access that field. What happens if you need to make a change to where that piece of state lives? You would now have to go change _every_ `useSelector` hook that references that value. So, in the same way that [we recommend using action creators to encapsulate details of creating actions](../style-guide/style-guide.md#use-action-creators), we recommend defining reusable selectors to encapsulate the knowledge of where a given piece of state lives. Then, you can use a given selector function many times in the codebase, anywhere that your app needs to retrieve that particular data.
```

Reply:
Just an idea - not sure if 'legal statement' would be confusing for translation purposes.